deep-security-agent/tasks/install-linux.yml

```diff
@@ -0,0 +1,149 @@
+  when: ansible_distribution == "Ubuntu"
+
+- name: Setting facts (SuSE) 
+  set_fact:
+    agent_relative_path: software/agent/SuSE_{{ ansible_distribution_major_version }}/{{ ansible_architecture }}
+  when: ansible_distribution == "SLES"
+
+- name: Setting facts (CloudLinux)
+  set_fact:
+    agent_relative_path: software/agent/CloudLinux_{{ ansible_distribution_major_version }}/{{ ansible_architecture }}
+  when: ansible_distribution == "CloudLinux"
+
+- name: Setting facts (OracleLinux)
+  set_fact:
+    agent_relative_path: software/agent/Oracle_OL{{ ansible_distribution_major_version }}/{{ ansible_architecture }}
+  when: ansible_distribution == "OracleLinux"
+
+- debug: 
+    msg: 
+      - "agent_local_path: {{ agent_local_path }}"
+      - "agent_relative_path: {{ agent_relative_path }}"
+
+# *********************************************************************
+# * Download Deep Security agent from Deep Security manager
+# *********************************************************************
+- debug:
+    msg: "Download from: https://{{ dsa_download_hostname }}:{{ dsa_download_port }}/{{ agent_relative_path }}/"
+
+- name: Download the Deep Security agent
+  get_url: 
+    url: "https://{{ dsa_download_hostname }}:{{ dsa_download_port }}/{{ agent_relative_path }}/" 
+    dest: "{{ agent_local_path }}"
+    validate_certs: no
+    timeout: 300
+    force: yes
+
+- stat:
+    path: "{{ agent_local_path }}"
+  register: agent_file
+
+- name: Check file size
+  debug:
+    msg: "Downloaded DSA installer file size : {{ agent_file.stat.size }} bytes"
+
+- fail:
+    msg:
+      - "Downloaded file size is not valid. There are two possiblities : 1. The OS platform is not supported. 2. Deep Security Manager don't provide agent software package installer for your OS platform."
+      - "Please verify that agent software package for your platform is available using this reference article https://help.deepsecurity.trendmicro.com/10/0/Get-Started/Install/import-agent-software.html?redirected=true&Highlight=download"
+  when: agent_file.stat.size == 0
+
+# *********************************************************************
+# * Install the agent
+# *********************************************************************
+# Both RPM and DPKG managers will do upgrade if downloaded package version is newer than installed. 
+# If versions are equals, no action. If older than installed, some managers will throw error.
+- name: Install the Deep Security agent (yum)
+  yum: 
+    name: "{{ agent_local_path }}"
+    state: present
+  when: ansible_pkg_mgr == "yum"
+  become: true
+
+- name: Install the Deep Security agent (apt)
+  apt: 
+    deb: "{{ agent_local_path }}"
+    state: present
+  when: ansible_pkg_mgr == "apt"
+  become: true
+
+- name: Install the Deep Security agent (zypper)
+  zypper: 
+    name: "{{ agent_local_path }}"
+    state: present
+    disable_gpg_check: yes
+  when: ansible_pkg_mgr == "zypper"
+  become: true
+
```

Comment:
Missing install process for "pkg5"

Reply:
Because we don't official support our playbook for Solaris, won't add it.